ldelf/ta_elf.c

```diff
@@ -733,6 +736,7 @@ static void populate_segments(struct ta_elf *elf)
 				if (res)
 					err(res, "sys_copy_from_ta_bin");
 			} else {
+				assert(filesz == memsz);
```

Comment:
assert here or return an error?

Reply:
I added this for debug purpose, but I don't see any harm in returning error instead. I'll update.